### DISCUSSION TITLE
Is there a way to omit a package or library from being minified or ignored by SWC?

### DISCUSSION THREAD

**Discussion by timndichu** [swc-project/swc#7175] [Q&A]
Hello,
I am building an audio chatroom in Next.js. For the chatting and calling features, I am using [Agora](https://www.agora.io/en/) that makes use of the ```agora-rtm-sdk``` [agora-rtm-sdk](https://www.npmjs.com/package/agora-rtm-sdk) and ```agora-rtc-sdk-ng``` [agora-rtc](https://www.npmjs.com/package/agora-rtc-sdk-ng) packages.

The issue comes in when I deploy to Vercel or any other hosting platform.

The ```agora-rtm-sdk``` package is not called at any point in my code, however the ```agora-rtc-sdk-ng``` works well

After struggling for days, I figured out the issue was in the bundling using [SWC][1].

This is my next.config.js file:

```
/** @type {import('next').NextConfig} */

const withTM = require('next-transpile-modules')([
  '@mui/material',
  '@mui/system',
  'agora-rtm-sdk',
  'agora-rtc-sdk-ng',
]);

module.exports = withTM({
  reactStrictMode: true,
  swcMinify: true, //However setting swcMinify: true causes rtm to not work
  images: {
    domains: ['mydomainname.cdn.digitaloceanspaces.com', 'localhost'],
  },
  compiler: {
    styledComponents: true,
  },

  webpack: (config) => {
    config.resolve.alias = {
      ...config.resolve.alias,
      '@mui/styled-engine': '@mui/styled-engine-sc',
    };
    config.optimization.splitChunks.cacheGroups = {
      common: {
        name: 'common',
        chunks: 'all',
      },
    };
    return config;
  },
});

```

So I got the app to work by setting ```swcMinify:false``` <br>
However, this was just a band-aid solution and makes the app slow.

How can I omit the ```agora-rtm-sdk``` from being minified while minifying the rest of the app or how can I make sure that my package is not being "swallowed" by SWC?<br><br>
Any help will be appreciated

  [1]: https://swc.rs/docs/getting-started

**Comment by kdy1** [comment]
There's no way to exclude it, but if the swc minifier breaks your package, please file an issue

**Reply by timndichu** [reply]
Thanks for your reply,
I'll file an issue

**Comment by thewh1teagle** [comment]
Any update? maybe its possible now?
